app/controllers/sessions_controller.rb

```diff
@@ -8,19 +7,21 @@ def destroy
     redirect_to root_url
   end
 
-  def create
+  def authsch_login
     raw_user = request.env['omniauth.auth']['extra']['raw_info']
     redirect_url = session[:redirect_url] || root_path
     user = User.find_by(auth_sch_id: raw_user['internal_id'])
     if user
       session[:user_id] = user.id
+      user.create_random_password if user.password_digest.nil?
```

Comment:
Ezt en inkabb kiraknam egy migracioba. Igy elhagyhato a User create_random_password methodja es amugy is csak egyszer kell lefuttatni minden meglevo usernel.

Reply:
Igaz, jobb lenne kirakni. Szerintem egy rake task-ban jobb helye lenne.